accepted/00070-akka.md

```diff
@@ -0,0 +1,139 @@
+
+All of the above was implemented (as proof-of-concept) as part of [this HackWeek PR](https://github.com/uyuni-project/uyuni/pull/1903)  (excluding commits after February 15, which deal with Akka Clustering) which can serve as a reference for implementation details.
+
+Some shortcuts have been taken during HackWeek and will not be acceptable for proper implementation, particularly:
+
+ - unit tests have been arbitrarily disabled or left broken
+   - in particular, when Message formats were significantly altered from the original code
+ - new methods/classes do not have new accompanying unit tests
+ - logging is incomplete
+ - code style of Akka-related parts follows the Akka coding conventions, which are different from the CheckStyle rules currently applied. We will either need to adapt new code to the existing conventions or relax them/create different rules
+ - error handling was not tested thoroughly, might need rework
+ - the Cucumber integration testsuite was not run on the resulting code
+ - performance tests will have to be conducted to make sure there is no regression (and evaluate possible speedups)
+ - the reflections-based mechanism to find out workers will have to be considered and it might be removed in exchange for an explicit approach
+ - most but not all Messages were made `Serializable`, more refactoring might be needed to ensure this is always the case
+ - Akka tuning knobs were not all exposed, or exposed by reusing existing configuration variables. New variables might need to be introduced and properly documented
+ - ideas from [the normalisation PR](https://github.com/uyuni-project/uyuni/pull/1954) should probably be integrated
+
+5. move specific Taskomatic task facilities (eg. `ErrataCacheDriver`, `SSHPushDriver`) to Akka
+6. refactor Taskomatic itself to use Akka
+
+Last point is about having Quartz-based Taskomatic logic only caring aobut the triggering of jobs, and leave to Akka their execution. Basically when Taskomatic fires (eg. on cron line) it will send a message to an Actor which would then carry out the actual computation.
+
+## Following Steps
+
+Following points need further research when points 1.-6. are completed, and possibly need follow-up RFCs, still are reported here for completeness.
+
+### Move Taskomatic code in the Tomcat web app
+
+After all asynchronous computation is moved to Akka, it might make sense to consider moving all Taskomatic code out of the currently separated daemon snd into the main Tomcat web app.
+
+Architecturally, such a change would have pros and cons:
+ - pro: we could remove the internal Taskomatic XMLRPC API, eliminating substantial complexity and the synchronization/availability problems we currently face (eg. [the race condition in `MinionActionExecutor`](https://github.com/uyuni-project/uyuni/blob/4.1-alpha1/java/code/src/com/redhat/rhn/taskomatic/task/MinionActionExecutor.java#L82-L102), as per [downstream issue 10657](https://github.com/SUSE/spacewalk/issues/10657))
```

Comment:
The `feature` of unauthenticated requests to the taskomatic API was really used by python scripts.
We would need to find a new way of connecting to it.

Reply:
Will add a note, however, keeping the endpoint is not a problem (we just need to make sure it ends up using the same TCP port).

Reply:
Fixed

Reply:
I was not sure if this is working with what you have in mind.